vaadin-rich-text-editor-flow-parent/vaadin-rich-text-editor-flow/src/main/java/com/vaadin/flow/component/richtexteditor/RichTextEditor.java

```diff
@@ -270,13 +270,17 @@ private String getDeltaValue() {
     }
 
     static String sanitize(String html) {
-        return org.jsoup.Jsoup.clean(html,
+        org.jsoup.nodes.Document.OutputSettings settings = new org.jsoup.nodes.Document.OutputSettings();
+        settings.prettyPrint(false);
```

Comment:
This doesn't seem to fix the [original issue](https://github.com/vaadin/flow-components/issues/5533). You can try it with the following:

```java
editor.asDelta().setValue(
                "[{\"attributes\":{\"bold\":true},\"insert\":\"Line 1\"},{\"insert\":\"\\n" + //
                                "\\tIndent 1\\n" + //
                                "\\t\\tIndent 2\\n" + //
                                "Last line with\\textra         spaces and a \\ttab\\n" + //
                                "\\n" + //
                                "\"}] ");
```

The above will still produce:
![Screenshot 2023-10-04 at 11 55 41](https://github.com/vaadin/flow-components/assets/1222264/e18d7e62-dfc9-47e8-9593-c8e9da8d4a62)

In fact, the result is the same even if you return the given `html` as such from the `sanitize` method.

Reply:
I need to dig this deeper. It may be that the fix needs changes in web component as well. Or if the white spaces are stripped also in Quill. If the later, we are stuck. If it is not the trouble maker, i.e. issue in our code, this Java side change is needed anyway, but if it is Quill related, this change is meaningless.